xbmc/addons/AddonManager.cpp

```diff
@@ -514,22 +515,25 @@ bool CAddonMgr::GetInstallableAddons(VECADDONS& addons, const TYPE &type)
   CSingleLock lock(m_critSection);
 
   // get all addons
-  VECADDONS installableAddons;
-  if (!m_database.GetRepositoryContent(installableAddons))
+  if (!m_database.GetRepositoryContent(addons))
     return false;
 
   // go through all addons and remove all that are already installed
-  for (const auto& addon : installableAddons)
-  {
-    // check if the addon matches the provided addon type
-    if (type != ADDON::ADDON_UNKNOWN && addon->Type() != type && !addon->IsType(type))
-      continue;
 
-    if (!CanAddonBeInstalled(addon))
-      continue;
+  addons.erase(std::remove(addons.begin(), addons.end(),
+    [&](const AddonPtr& addon)
```

Comment:
Not sure I agree with explicit capture being a good habit *everywhere*. When the lambda outlives the scope, for sure, because then horrible things can happen. But for things like std::remove it's perfectly safe.